### DISCUSSION TITLE
Homebrew Tap

### DISCUSSION THREAD

**Discussion by mhanberg** [zk-org/zk#64] [Share]
To make it easier for me to install zk on my various computers, I've made a homebrew tap

https://github.com/mhanberg/homebrew-zk

It currently has bottles (pre-compiled binaries) for x86-64 macOS Catalina (should still work on Big Sur) and x86-64 Linux, but you can also use it to compile from source.

```shell
$ brew install mhanberg/zk/zk
$ brew install --build-from-source mhanberg/zk/zk
```

## arm64

It looks like GitHub Actions is introducing macOS 11 runners next week, so I should be able to get this producing macOS Big Sur binaries as well as arm64 binaries for Mac and Linux

**Comment by mickael-menu** [comment]
Nice, thanks! Is it something you will need to update with every new version of `zk`?

Maybe it would make sense to have it in the official repo, so I can make sure the Go flags are always up to date.

**Reply by mhanberg** [reply]
> Is it something you will need to update with every new version of zk?

I think so

> Maybe it would make sense to have it in the official repo, so I can make sure the Go flags are always up to date.

If you want to take on the maintenance, sure! I think tho that for the tap to work, it needs to be in a repo that is prefixed by `homebrew-`, hence the name `homebrew-zk`.

Ideally we would get this merged into homebrew core and remove the need to have a custom tap.

**Reply by mickael-menu** [reply]
Okay, let's keep it on your repo for now, and if at some point you can't maintain it anymore we can migrate it here. Thanks again 👍

**Comment by mhanberg** [comment]
I finally figured out how to update the homebrew formula

v0.7.0 is available now with mac and linux bottles https://github.com/mhanberg/homebrew-zk/releases/tag/zk-0.7.0

**Reply by mhanberg** [reply]
I'm going to set up another workflow to listen for tags from the zk repo and automatically do this

**Reply by mickael-menu** [reply]
Nice, that would be convenient

**Comment by mhanberg** [comment]
I submitted a PR to add zk to the homebrew-core forumulas https://github.com/Homebrew/homebrew-core/pull/91670.

If that is merged (I assume it will be), I think you can probably set up a github actions automation to create a PR to update the formula when you make a new release.

**Reply by mickael-menu** [reply]
Sounds great, thanks for taking the time to look into that!
